### PR TITLE
[FLASK] Update UI related to MetaMask Snaps Platform trademark

### PR DESCRIPTION
**Pull request by david0xd**
Fixes: https://github.com/MetaMask/MetaMask-planning/issues/358

## Explanation
This PR is following the text and design updates following the requests from https://github.com/MetaMask/MetaMask-planning/issues/358.

Only trademark update is required on a Settings page for Snaps. Other places are matching the rule explained in https://github.com/MetaMask/MetaMask-planning/issues/358.

https://github.com/MetaMask/MetaMask-planning/issues/358 was also requesting better design of the top two lines of text on the Snaps list, but these are removed according to the new design proposal for this page which should go after this work is done: https://github.com/MetaMask/snaps-monorepo/issues/1170 (https://www.figma.com/file/diidnCHx9FXj5LDiA5DZZr/Snaps-features?node-id=55-11895&t=i31mw6uZ1dow4i0h-0).

### Before
Screenshot 2023-03-16 at 17 56 54
Screenshot 2023-03-16 at 17 57 48

### After
Screenshot 2023-03-21 at 13 02 14
New_Screenshot 2023-03-21 at 13 03 17

## Manual Testing Steps
1. Open MetaMask extension popup or fullscreen view.
2. Go to the Settings > Snaps.
3. Install or remove Snap depending on which part is tested.

## Pre-merge author checklist

- [x] I've clearly explained:
  - [x] What problem this PR is solving
  - [x] How this problem was solved
  - [x] How reviewers can test my changes
- [x] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

### CODE REVIEW

File: ui/pages/settings/flask/snaps-list-tab/snap-list-tab.js [hunk 1]
```diff
@@ -85,12 +81,28 @@ const SnapListTab = () => {
...
+            </Typography>
+            <Typography variant={TypographyVariant.H4}>
+              <Text>
+                {`${t('noSnaps')} `}
+                <ButtonLink
+                  size={Size.inherit}
+                  href="https://metamask.io/snaps/"
+                  target="_blank"
+                >
+                  {t('noSnapsLearnMoreLinkText')}
```
[the remark this answers is not in the document]
Comment (david0xd) [reply ui/pages/settings/flask/snaps-list-tab/snap-list-tab.js]: Yes, updated.

**Review by rekmarks — CHANGES_REQUESTED** [review]
We should use "MetaMask Snaps" to refer to the platform. We should try to avoid using those words to refer to snaps collectively.

File: app/_locales/en/messages.json [hunk 2]
```diff
@@ -1863,6 +1863,9 @@
   "lastSold": {
     "message": "Last sold"
   },
+  "learnAboutSnaps": {
+    "message": "Learn about MetaMask Snaps"
+  },
```
Comment (rekmarks) [inline app/_locales/en/messages.json:1868]:
```suggestion
  "exploreWithSnaps": {
    "message": "Explore web3 with MetaMask Snaps"
  },
```

File: ui/pages/settings/flask/snaps-list-tab/snap-list-tab.js [hunk 3]
```diff
@@ -82,15 +81,58 @@ const SnapListTab = () => {
...
+            alignItems={AlignItems.flexEnd}
+            paddingLeft={4}
+            paddingRight={4}
+            paddingBottom={4}
           >
-            <span>{t('noSnaps')}</span>
-          </Typography>
+            <BannerTip
+              logoType={BannerTipLogoType.Greeting}
+              title={t('learnAboutSnaps')}
```
Comment (rekmarks) [inline ui/pages/settings/flask/snaps-list-tab/snap-list-tab.js:124]:
```suggestion
              title={t('exploreWithSnaps')}
```

**Review by rekmarks — DISMISSED** [review]
LGTM!

**Review by FrederikBolding — APPROVED** [review]

**Review by rekmarks — APPROVED** [review]

**Pull request merged**